libraries/core_libs/network/src/tarcap/packets_handlers/votes_sync_packet_handler.cpp

```diff
@@ -24,10 +24,10 @@ void VotesSyncPacketHandler::validatePacketRlpFormat([[maybe_unused]] const Pack
 void VotesSyncPacketHandler::process(const PacketData &packet_data, const std::shared_ptr<TaraxaPeer> &peer) {
   const auto next_votes_count = packet_data.rlp_.itemCount();
   if (next_votes_count == 0 || next_votes_count > kMaxVotesInPacket) {
-    LOG(log_er_) << "Receive " << next_votes_count << " next votes from peer " << packet_data.from_node_id_
-                 << ". The peer may be a malicious player, will be disconnected";
-    disconnect(packet_data.from_node_id_, dev::p2p::UserReason);
-    return;
+    std::ostringstream err_msg;
+    err_msg << "Receive " << next_votes_count << " next votes from peer " << packet_data.from_node_id_
+            << ". The peer is a malicious player, will be disconnected";
+    throw MaliciousPeerException(err_msg.str());
```

Comment:
This check seems like it could be rather inside validatePacketRlpFormat because it is related to rlp format

Reply:
fixed